rich/pretty.py

```diff
@@ -148,6 +148,18 @@ def _ipy_display_hook(
                 if repr_result is not None:
                     return  # Delegate rendering to IPython
 
+        # When in a Jupyter notebook let's avoid the display of some specific classes,
+        # as they result in the rendering of useless and noisy lines such as `<Figure size 432x288 with 1 Axes>`.
+        # What does this do?
+        # --> if the class has "matplotlib.artist.Artist" in its hierarchy for example, we don't render it.
+        classes_hierarchy_fully_qualified_names = {
+            # With this `[8:-2]` we convert  "<class 'pkg.Something'>" to  "pkg.Something":
+            str(class_)[8:-2]
```

Comment:
Sorry, that's not a reliable mechanism. It may well work most of the time, but there is no guarantee that will be the case for for all classes.

Consider using `__module__` which I *think* gets the fully qualified path to the module.

We also do not want to make any assumptions about the object being printed. There is no guarantee it is a class instance, and therefore has a `__class__`

Reply:
Off the top of your head, would you have an example of a object that doesn't have a `__class__`, so I can test another way to manage this with various types of exotic objects?
(because if an object doesn't have such an attribute I'll likely struggle to get the part of its fully qualified name that comes after its module, so I would like to see what's possible for such objects :slightly_smiling_face: )

We would also have to find a way to get qualified names of objects that don't have a `__module__`: 
![Screenshot from 2022-06-29 15-31-10](https://user-images.githubusercontent.com/722388/176462748-86e7330f-d3fa-4619-9988-8863e1a1a9a2.png)

Reply:
Anything with `__getattribute__` defined. Which happens more than you might think. Usually from ORMs or other magic classes.

You're absolutely correct that `__module__` may not exist. You can't assume anything about objects, you will probably have to resort to `getattr`